medmodels/data_synthesis/synthesizer.pyi

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+from typing import Dict, Optional
+
+import torch
+from torch import nn
+
+from medmodels.data_synthesis.builder import SynthesizerBuilder
+from medmodels.data_synthesis.synthesizer_model import SynthesizerModel
+from medmodels.medrecord.medrecord import MedRecord
+
+class Synthesizer(nn.Module):
+    preprocessor: nn.Module
+    postprocessor: nn.Module
+    device: torch.device
+
+    def __init__(
+        self,
+        preprocessor: nn.Module,
+        postprocessor: nn.Module,
+    ) -> None: ...
+    @classmethod
+    def builder(cls) -> SynthesizerBuilder: ...
+    def preprocess(self, medrecord: MedRecord) -> MedRecord: ...
+    def fit(
+        self, medrecord: MedRecord, save_directory: Optional[Path]
+    ) -> SynthesizerModel: ...
+    def fit_from(
+        self,
+        medrecord: MedRecord,
+        save_directory: Optional[Path],
+        **kwargs: Dict[str, Path],
+    ) -> SynthesizerModel: ...
+    def save_model(self, path: Path) -> None: ...
+    def load_model(self, path: Path) -> SynthesizerModel: ...
```

Comment:
shoul probably also be added to the Synthesizer builder

Reply:
Added it to the MTGANBuilder (medmodels.data_synthesis.mtgan.builder), since it would make not that much sense to add it to the SynthesizerBuilder (the idea is loading some parameters and not a whole model, and that is difficult without knowing which Synthesizer we are using).

If you do not agree, please let me know with another comment.